android/app/src/main/kotlin/com/example/sloth/AndroidSignerPlugin.kt

```
@@ -0,0 +1,546 @@
+        val intent = Intent(Intent.ACTION_VIEW, Uri.parse("$NOSTRSIGNER_SCHEME:${Uri.encode(eventJson)}"))
+        intent.`package` = packageName
+        intent.putExtra("type", "sign_event")
+        intent.putExtra("id", id)
+        if (currentUser.isNotEmpty()) {
+            intent.putExtra("current_user", currentUser)
+        }
+        intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP or Intent.FLAG_ACTIVITY_CLEAR_TOP)
+
+        try {
+            currentActivity.startActivityForResult(intent, REQUEST_CODE_SIGNER)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)